### PR TITLE
[UNDER POWERCUT TESTING] os/fs/driver/mtd: Remove additional CRC check of data read from flash

### PR DESCRIPTION
**Pull request by amogh-samsung**
- We do not need a CRC check for verification at read time if read back and verify happens at write time.
- In our focus area of jSmartFs (Journaling enabled), read back and verify always happens at write time,
 which gives us the option to avoid this unnecessarily time taking overhead.

Signed-off-by: Amogh Hassija <a.hassija@samsung.com>

### CODE REVIEW

File: os/fs/driver/mtd/smart.c [hunk 1]
```diff
@@ -4408,26 +4408,6 @@ static int smart_readsector(FAR struct smart_struct_s *dev, unsigned long arg)
 		fdbg("Error reading phys sector %d\n", physsector);
 		return -EIO;
 	}
-#if SMART_STATUS_VERSION == 1
```
Comment (thapav) [inline os/fs/driver/mtd/smart.c:4411]: Can you please specify the improvement percentage in reading without the CRC check in the commit details ?

Comment (thapav) [inline os/fs/driver/mtd/smart.c:4411]: What about the cases where journaling is not enabled and/or the readback wont happen while writing to flash?

Comment (amogh-samsung) [reply os/fs/driver/mtd/smart.c:4411]: I'll add a TODO comment for the same
We develop SmartFs with journaling primarily as a robust and thorough solution that guarantees FS integrity.
Hence cases where journaling is disabled are possible to build but do not majorly concern us

Comment (Taejun-Kwon) [reply os/fs/driver/mtd/smart.c:4411]: I think you can simply update it instead of TODO comment. 3~5 lines of code will be enough

**Comment by Taejun-Kwon** [comment]
Major changes, powercut test should be done

**Review by Taejun-Kwon — APPROVED** [review]

**Pull request merged**